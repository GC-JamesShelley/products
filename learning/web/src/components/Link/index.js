import React from "react"
import { Link as GatsbyLink } from "gatsby"
import styled from "styled-components"
import { anchorColour, mhraBlue } from "../../utils/colors"

const GatsbyLinkStyled = styled(GatsbyLink)`
  color: ${anchorColour};
  text-decoration: none;
  &:hover,
  &:active {
    color: ${mhraBlue};
    text-decoration: underline;
  }
`
const Footnote = styled.a`
  color: ${anchorColour};
  &:hover,
  &:active {
    color: ${mhraBlue};
    text-decoration: underline;
  }
`

const Link = ({ children, to, activeClassName, partiallyActive, ...other }) => {
  // Tailor the following test to your environment.
  // This example assumes that any internal link (intended for Gatsby)
  // will start with exactly one slash, and that anything else is external.
  const internal = /^\/(?!\/)/.test(to)
  const footnote = /^#(?!\/)/.test(to)
  // Use Gatsby Link for internal links, and <a> for others
  if (internal) {
    return (
      <GatsbyLinkStyled
        to={to}
        activeClassName={activeClassName}
        partiallyActive={partiallyActive}
        {...other}
      >
        {children}
      </GatsbyLinkStyled>
    )
  } else if (footnote) {
    return (
      <Footnote href={to} {...other}>
        {children}
      </Footnote>
    )
  }
  return (
    <a href={to} {...other}>
      {children}
    </a>
  )
}

export default Link
