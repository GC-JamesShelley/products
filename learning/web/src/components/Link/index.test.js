import React from "react"
import Link from "./index"
import { shallow } from "enzyme"

const location = { pathname: "test" }

describe(Link, () => {
  it("should render 🍪", () => {
    const component = shallow(
      <Link
        children={<> </>}
        to={"infinity"}
        activeClassName={"test"}
        partiallyActive={true}
      />
    )
    expect(component).toMatchSnapshot()
  })
})
