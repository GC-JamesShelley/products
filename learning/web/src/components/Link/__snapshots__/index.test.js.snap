// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render 🍪 1`] = `
<Styled(styled.a)
  href="infinity"
  rel="noopener noreferrer"
  target="_blank"
>
   
  <span>
    <FiExternalLink />
  </span>
</Styled(styled.a)>
`;
